apps/meteor/client/views/account/profile/AccountProfilePage.tsx

```diff
@@ -92,18 +92,24 @@ const AccountProfilePage = (): ReactElement => {
 	);
 
 	const handleDeleteOwnAccount = useCallback(async () => {
-		const handleConfirm = async (passwordOrUsername: string): Promise<void> => {
+		const handleConfirm = async (passwordOrUsername: string, setInputError: (message: string) => void ): Promise<void> => {
 			try {
 				await deleteOwnAccount({ password: SHA256(passwordOrUsername) });
 				dispatchToastMessage({ type: 'success', message: t('User_has_been_deleted') });
+				setModal(null);
 				logout();
 			} catch (error: any) {
 				if (error.error === 'user-last-owner') {
 					const { shouldChangeOwner, shouldBeRemoved } = error.details;
 					return handleConfirmOwnerChange(passwordOrUsername, shouldChangeOwner, shouldBeRemoved);
 				}
 
-				dispatchToastMessage({ type: 'error', message: error });
```

Comment:
I noticed you removed this toast dispatch, but what if there are other errors? Such as a network error or something else... I think we should probably keep this as the "last resort" error handling.